### ISSUE TITLE
Path localization with i18n and directory_indexes (v3.4.0)

### ISSUE THREAD

**Issue by sn3p** [middleman/middleman#1587]
In v3 `directory_indexes` should be activated AFTER `i18n` (see https://github.com/middleman/middleman/issues/1525).
But then the i18n `url_for` [helper](https://github.com/middleman/middleman/blob/v3-stable/middleman-core/lib/middleman-more/core_extensions/i18n.rb#L63-L88) will not translate paths. When `directory_indexes` is activated BEFORE `i18n` then paths are translated, but then `directory_indexes` isn't working.

Is this a knows issue and is there a workaround?

**Comment by tdreyno** [comment]
Yikes. Let me look

**Comment by sn3p** [comment]
Also the `url_for` [i18n helper](https://github.com/middleman/middleman/blob/v3-stable/middleman-core/lib/middleman-more/core_extensions/i18n.rb#L63-L88) doesn't handle fragments, anchors and query params, like the [core helper](https://github.com/middleman/middleman/blob/f5fe46cb593dcbe57baf870320ca3f8978ca6377/middleman-core/lib/middleman-core/util.rb#L191-L198) does.

```erb
<%= url_for("producten.html", locale: :de) %><br>
<%= url_for("products.html#test", locale: :de) %><br>
<%= url_for("products.html?test=1", locale: :de) %><br>
```
Outputs:
```html
de/produkte.html
products.html#test
products.html?test=1
```

**Comment by karland** [comment]
@tdreyno Any updates to this? Cheers.

**Comment by tdreyno** [comment]
Pretty sure this is almost entirely fixed on v4, other than #1798

**Comment by andreamoro** [comment]
I have been trying to use the url_for, on MiddleMan 4 changing the [examples](https://github.com/middleman/middleman/issues/1587#issuecomment-135469826) so to reflect my locale, but I can't get anything out other than the image path specified. Nothing else is add.

Is the url_for relative or depending by some internal setting like css_dir or images_dir?
